assets/app/features/upgrade/templates/landing-page_step3.partial.jade

```diff
@@ -0,0 +1,12 @@
+.upgrade(ng-class="{'active': upgradeLandingVm.mode.valid}")
+    .step-title.clearfix
+        crowbar-circle.pull-left(ng-class="{'active': upgradeLandingVm.mode.valid}") 3
```

Comment:
Directives are hard on the surface but once you get inside they are pretty nice concept to work with.

Take a look around here: https://github.com/crowbar/crowbar-ui/blob/master/assets/app/widgets/suse-lazy-spinner/suse-lazy-spinner.directive.js#L27 for some hints on how to handle custom attributes (by coincidence this one is also called 'active'). You generally need to create 'link' function which is called by angular when the directive is attached to specific DOM element. This is the only place where you get a DOM handle to work with. You can register handlers and watches here to define your directive's behavior.

Another place to look at is just above (https://github.com/crowbar/crowbar-ui/blob/master/assets/app/widgets/suse-lazy-spinner/suse-lazy-spinner.directive.js#L15). This is where you define "isolated scope" for the directive. This means that each instance of this directive will have it's own "interface" and will not rely on the parent scope.

Final piece is something you probably saw already: https://github.com/crowbar/crowbar-ui/blob/master/assets/app/features/upgrade/templates/landing-page.jade#L38 binding 'active' attribute of specific instance of the directive with some external model.